rabbitmq.go

```diff
@@ -192,8 +195,18 @@ func (m *RabbitMqClient) On(topicName string, consumerName string, handlerFunc M
 
 func (m *RabbitMqClient) Close() error {
 
+	if m.isClosed.Load() {
+		return nil
+	}
+
 	m.isClosed.Store(true)
 
+	for i := 0; i < (len(m.subscribers) * m.threadNum); i++ {
```

Comment:
@bintangsibarani we can call close(m.signaldone) instead of writing to this channel. Close will make the channel reader read an empty value of this channel, hence the loop will break.